csharp.html.markdown

```diff
@@ -1008,15 +1008,108 @@ on a new line! ""Wow!"", the masses cried";
             // x?.y will return null immediately if x is null; y is not evaluated
             => GenieName?.ToUpper();
     }
+
+    static class MagicService
+    {
+        private static bool LogException(Exception ex)
+        {
+            /* log exception somewhere */
+            return false;
+        }
+
+        public static bool CastSpell(string spell)
+        {
+            try
+            {
+                // Pretend we call API here
+                throw new MagicServiceException("Spell failed", 42);
+
+                // Spell succeeded
+                return true;
+            }
+            // Only catch if Code is 42 i.e. spell failed
+            catch(MagicServiceException ex) when (ex.Code == 42)
+            {
+                // Spell failed
+                return false;
+            }
+            // Suppose the API is quirky and sometimes returns a
+            // nonzero result code even if the spell succeeds
+            catch(MagicServiceException ex) when (ex.Code == 117)
```

Comment:
I'm not sure this case is needed – does it show any other feature the previous case doesn't show?